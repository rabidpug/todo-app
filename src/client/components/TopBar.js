import {
  Button,
  Icon,
  Layout,
  Menu,
} from 'antd';

import React from 'react';
import styles from './TopBar.scss';

const { Item, SubMenu, } = Menu;

const { Header, } = Layout;
const TopBar = ( {
  isSidebarCollapsed, toggleSideBar, actionMenuItems, selectedActionKeys,
} ) => {
  const menuItemMap = item =>
    item.subMenu ? (
      <SubMenu
        key={ item.key }
        title={
          <span>
            <Icon type={ item.icon } />
            <span>{item.label}</span>
          </span>
        }>
        {item.subMenu.map( menuItemMap )}
      </SubMenu>
    ) : (
      <Item
        action={ item.action }
        key={ item.key }>
        {item.component
          ? <item.component { ...item } />
          : (
            <span>
              <Icon type={ item.icon } />
              {item.label}
            </span>
          )}
      </Item>
    );

  return (
    <Header
      style={ { background : '#fff',
                padding    : 0, } }>
      <Button
        className={ styles.trigger }
        icon={ isSidebarCollapsed ? 'menu-unfold' : 'menu-fold' }
        onClick={ toggleSideBar }
      />
      {actionMenuItems && (
        <Menu
          className={ styles.topBarMenu }
          mode='horizontal'
          onClick={ ( { item, } ) => item.props.action && item.props.action() }
          selectedKeys={ typeof selectedActionKeys === 'object' ? selectedActionKeys : [ selectedActionKeys, ] }>
          {actionMenuItems.map( menuItemMap )}
        </Menu>
      )}
    </Header>
  );
};

export default TopBar;
