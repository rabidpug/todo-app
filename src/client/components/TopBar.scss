.topBarMenu {
  float: right;
}

.trigger {
  margin-left: 10px;
}
