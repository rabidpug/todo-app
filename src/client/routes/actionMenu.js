import FilterLink from 'Containers/FilterLink';
import HomeTopBar from 'Containers/HomeTopBar';
import ShowModal from 'Containers/ShowModal';
import TodoTopBar from 'Containers/TodoTopBar';
import { VisibilityFilters, } from 'Store/todoVisibility/types';

const actionMenu = [
  {
    actionMenuItems: [
      {
        component : ShowModal,
        icon      : 'plus',
        key       : '1',
        label     : 'Add Todo',
      },
      {
        icon    : 'eye-o',
        key     : '2',
        label   : 'Show...',
        subMenu : [
          {
            component : FilterLink,
            filter    : VisibilityFilters.SHOW_ALL,
            icon      : 'bars',
            key       : VisibilityFilters.SHOW_ALL,
            label     : 'All',
          },
          {
            component : FilterLink,
            filter    : VisibilityFilters.SHOW_ACTIVE,
            icon      : 'exclamation-circle-o',
            key       : VisibilityFilters.SHOW_ACTIVE,
            label     : 'Active',
          },
          {
            component : FilterLink,
            filter    : VisibilityFilters.SHOW_COMPLETED,
            icon      : 'check',
            key       : VisibilityFilters.SHOW_COMPLETED,
            label     : 'Completed',
          },
        ],
      },
    ],
    component : TodoTopBar,
    exact     : true,
    path      : '/todo',
  },
  {
    actionMenuItems : [],
    component       : HomeTopBar,
    path            : '*',
  },
];

export default actionMenu;
