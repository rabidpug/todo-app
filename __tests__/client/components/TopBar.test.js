import React from 'react';
import TopBar from 'Components/TopBar';
import actionMenu from 'Routes/actionMenu';
import { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';

describe(
  'TopBar', () => {
    it(
      'Should render the TopBar', () => {
        const [ { actionMenuItems, }, ] = actionMenu.filter( menu => menu.path === '/todo' );
        const wrapper = shallow( <TopBar actionMenuItems={ actionMenuItems } /> );

        expect( shallowToJson( wrapper ) ).toMatchSnapshot();
      }
    );
  }
);
