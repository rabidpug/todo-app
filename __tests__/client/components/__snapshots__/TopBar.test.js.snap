// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Should render the TopBar 1`] = `
<Adapter
  style={
    Object {
      "background": "#fff",
      "padding": 0,
    }
  }
>
  <Button
    className="trigger"
    ghost={false}
    icon="menu-fold"
    loading={false}
    prefixCls="ant-btn"
  />
  <Menu
    className="topBarMenu"
    mode="horizontal"
    onClick={[Function]}
    prefixCls="ant-menu"
    selectedKeys={
      Array [
        undefined,
      ]
    }
    theme="light"
  >
    <MenuItem
      key="1"
    >
      <Connect(Link)
        component={[Function]}
        icon="plus"
        key="1"
        label="Add Todo"
      />
    </MenuItem>
    <SubMenu
      key="2"
      title={
        <span>
          <Icon
            type="eye-o"
          />
          <span>
            Show...
          </span>
        </span>
      }
    >
      <MenuItem
        key="SHOW_ALL"
      >
        <Connect(Link)
          component={[Function]}
          filter="SHOW_ALL"
          icon="bars"
          key="SHOW_ALL"
          label="All"
        />
      </MenuItem>
      <MenuItem
        key="SHOW_ACTIVE"
      >
        <Connect(Link)
          component={[Function]}
          filter="SHOW_ACTIVE"
          icon="exclamation-circle-o"
          key="SHOW_ACTIVE"
          label="Active"
        />
      </MenuItem>
      <MenuItem
        key="SHOW_COMPLETED"
      >
        <Connect(Link)
          component={[Function]}
          filter="SHOW_COMPLETED"
          icon="check"
          key="SHOW_COMPLETED"
          label="Completed"
        />
      </MenuItem>
    </SubMenu>
  </Menu>
</Adapter>
`;
